custom_components/tuya_local/devices/soho_air_purifier.yaml

```diff
@@ -0,0 +1,122 @@
+name: Air Purifier
+primary_entity:
+  entity: fan
+  translation_only_key: fan_with_presets
+  dps:
+    - id: 1
+      name: switch
+      type: boolean
+    - id: 3
+      name: preset_mode
+      type: string
+      mapping:
+        - dps_val: auto
```

Comment:
Instead of defining new presets for auto and manual, you could use the existing "smart" and "normal" presets (the existing ones were copied from predefined presets in the standard HA fan component, which hasn't been translated yet)